js_sdk/end_idle_engagements.js

```diff
@@ -0,0 +1,32 @@
+// Reference:
+// https://js-sdk-docs.salemove.com/class/Engagement.html
+// Running the code snippet below before starting an engagement
+// allows to track Visitor activity to end the ongoing engagement
+// when Visitor has been idle (no mouse or keyboard event recorded)
+// for some given amount of time
+
+sm.getApi({version: 'v1'}).then(function(salemove) {
+  var onEngagementStarted = function(engagement) {
+    var idleTime = 0;
+    var timerIncrement = function() {
+      idleTime = idleTime + 1;
+      if (idleTime >= 29) {
+        engagement.end();
+      }
+    }
+    var idleInterval = setInterval(timerIncrement, 1000);
+    $(window).mousemove(function(e) {
```

Comment:
Why do you want to use jquery? We can't expect that client is using jquery. Rather write it in plain JS.